tests/test_qaoa.py

```diff
@@ -134,3 +135,74 @@ def test_xy_mixer_output(self, graph, target_hamiltonian):
         assert mixer_coeffs == target_coeffs
         assert mixer_ops == target_ops
         assert mixer_wires == target_wires
+
+class TestCostHamiltonians:
+    """Tests that the cost Hamiltonians are being generated correctly"""
+
+    def test_maxcut_error(self):
+        """Tests that the MaxCut Hamiltonian throws the correct error"""
+
+        graph = [(0, 1), (1, 2)]
+
+        with pytest.raises(ValueError, match=r"nput graph must be a nx\.Graph"):
+            output = qaoa.MaxCut(graph)
+
+    @pytest.mark.parametrize(
+        ("graph", "target_hamiltonian"),
+        [
+            (
+                Graph([(0, 1), (1, 2)]),
+                qml.Hamiltonian(
+                    [-0.5, 0.5, -0.5, 0.5],
+                    [
+                        qml.Identity(0) @ qml.Identity(1),
+                        qml.PauliZ(0) @ qml.PauliZ(1),
+                        qml.Identity(1) @ qml.Identity(2),
+                        qml.PauliZ(1) @ qml.PauliZ(2),
+                    ],
+                ),
+            ),
+            (
+                Graph((np.array([0, 1]), np.array([1, 2]), np.array([0, 2]))),
+                qml.Hamiltonian(
+                    [-0.5, 0.5, -0.5, 0.5, -0.5, 0.5],
+                    [
+                        qml.Identity(0) @ qml.Identity(1),
+                        qml.PauliZ(0) @ qml.PauliZ(1),
+                        qml.Identity(0) @ qml.Identity(2),
+                        qml.PauliZ(0) @ qml.PauliZ(2),
+                        qml.Identity(1) @ qml.Identity(2),
+                        qml.PauliZ(1) @ qml.PauliZ(2),
+                    ],
+                ),
+            ),
+            (
+                graph,
+                qml.Hamiltonian(
+                    [-0.5, 0.5, -0.5, 0.5],
+                    [
+                        qml.Identity(0) @ qml.Identity(1),
+                        qml.PauliZ(0) @ qml.PauliZ(1),
+                        qml.Identity(1) @ qml.Identity(2),
+                        qml.PauliZ(1) @ qml.PauliZ(2),
+                    ],
+                ),
+            ),
+        ],
+    )
```

Comment:
Minor comment:
It might be nice to build this up earlier in `TestCostHamiltonians` and then pass to `parametrize()`, e.g.,

- Create graphs
- Create hamiltonians
- Create list of pairs

^ Just might help with readability.